### ISSUE TITLE
Reproduce degree of urbanisation

### ISSUE THREAD

**Issue by jipclaassens** [ObjectVision/2BURP#29]
https://ghsl.jrc.ec.europa.eu/degurbaDefinitions.php

Note that the SMOD analysis is performed in MollWeide. Our model is in LatLong, so proximity will be different.

**Comment by jipclaassens** [comment]
- Urban centre (or high-density cluster
  - contiguous cell with density >= 1500 person per km2
  - total pop >= 50,000 persons
  - gaps are filled, and edges smoothed
  - if needed, cells with built-up share >= 50% are added.
- Urban cluster (or moderate density clusters)
  - contiguous cells with density >= 300 persons per km2
  - total pop >= 5,000 persons
  - urban centres are subsets of urban clusters
- Rural grid cells (mostly low density cells)
  - cell that do not belong to urban clusters
  - most cells < 300 persons per km2
  - some cells have higher density, but not part of cluster with enough population

**Comment by jipclaassens** [comment]
@cjacobscrisioni what do you think?

![image](https://github.com/ObjectVision/2BURP/assets/48675272/3bd81398-6789-4974-823b-81652ffcf4e3)

**Comment by cjacobscrisioni** [comment]
Seems plausible, well done! Which year did you base this on? 

We'll have a team meeting on the 25th. Probably good to ask the team for anybody who can compare this output with the 'formal' GHSL degree of urbanisation results.

**Comment by jipclaassens** [comment]
I've used the year 2020.

Yes, will do!

There are some differences between my version and the formal version:
1. Analysis in WGS84 vs Mollweide: This has an impact on the filtering/gap filling and on the distance-related measures. I've now used a kernel with a radius of about 3 cells, but of course, it isn't always 3 km.
2. I need a definition for' on land'. I've now used the version we use in the model from the coast data set.
3. I truncate the population density as discussed in https://github.com/ObjectVision/2BURP/issues/34
